scripts/doc_checker.py

```diff
@@ -0,0 +1,559 @@
+    elif "# noqa" in noqa_str:
+        noqa_checks = ["all"]
+    else:
+        noqa_checks = []
+    return [check.strip() for check in noqa_checks]
+
+
+# code snippet from numpydoc
+def validate_object(import_path: str) -> bool:
+    """
+    Check docstrings of an entity that can be imported.
+
+    Parameters
+    ----------
+    import_path : str
+        Python-like import path.
+
+    Returns
+    -------
+    is_successfull : bool
+        Return True if all checks are successful.
+    """
+    from numpydoc.validate import validate
+
+    is_successfull = True
+    doc = Docstring(import_path)
+    results = validate(import_path)
+    results = validate_modin_error(doc, results)
+    noqa_checks = get_noqa_checks(doc)
+    for err_code, err_desc in results["errors"]:
+        if (
+            err_code not in NUMPYDOC_BASE_ERROR_CODES
+            and err_code not in MODIN_ERROR_CODES
+        ) or skip_check_if_noqa(doc, err_code, noqa_checks):
+            continue
+        is_successfull = False
+        logging.error(
+            ":".join([import_path, str(results["file_line"]), err_code, err_desc])
+        )
+    return is_successfull
+
+
+def numpydoc_validate(path: pathlib.Path) -> bool:
+    """
+    Perform numpydoc checks.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        Filename or directory path for check.
+
+    Returns
+    -------
+    is_successfull : bool
+        Return True if all checks are successful.
+    """
+    is_successfull = True
+
+    if path.is_file():
+        walker = ((str(path.parent), [], [path.name]),)
+    else:
+        walker = os.walk(path)
+
+    for root, _, files in walker:
+        if "__pycache__" in root:
+            continue
+        for _file in files:
+            if not _file.endswith(".py"):
+                continue
+
+            current_path = os.path.join(root, _file)
+            # get importable name
+            module_name = current_path.replace("/", ".").replace("\\", ".")
+            # remove ".py"
+            module_name = os.path.splitext(module_name)[0]
+
+            with open(current_path) as fd:
+                file_contents = fd.read()
+
+            # using static parsing for collecting module, functions, classes and their methods
+            module = ast.parse(file_contents)
+
+            def is_public_func(node):
+                return isinstance(node, ast.FunctionDef) and (
+                    not node.name.startswith("__") or node.name.endswith("__")
+                )
+
+            functions = [node for node in module.body if is_public_func(node)]
+            classes = [node for node in module.body if isinstance(node, ast.ClassDef)]
+            methods = [
+                f"{module_name}.{_class.name}.{node.name}"
+                for _class in classes
+                for node in _class.body
+                if is_public_func(node)
+            ]
+
+            logging.info(f"NUMPYDOC OUTPUT FOR {current_path} - CAN BE EMPTY")
```

Comment:
As for me, output like 
```
INFO:NUMPYDOC OUTPUT FOR example.py  CAN BE EMPTY
```
confuses me a little. I would prefer something like this:
```
INFO:NUMPYDOC OUTPUT FOR example.py  SUCCESSFUL
```

Reply:
Now the message appears only in case of an error.
```python
INFO:PYDOCSTYLE OUTPUT FOR scripts\doc_checker.py
ERROR:scripts\doc_checker.py:341 in public function `numpydoc_validate`:
        D103: Missing docstring in public function

INFO:NUMPYDOC OUTPUT FOR scripts\doc_checker.py
ERROR:scripts.doc_checker.numpydoc_validate:341:GL08:The object does not have a docstring
ERROR:scripts.doc_checker.pydocstyle_validate:397:PR01:Parameters {'add_ignore'} not documented
ERROR:scripts.doc_checker.pydocstyle_validate:397:PR02:Unknown parameters {'add_ignore: List[int]'}
ERROR:scripts.doc_checker.pydocstyle_validate:397:PR10:Parameter "add_ignore" requires a space before the colon separating the parameter name and type
ERROR:INVALID DOCUMENTATION FOUND
```

Reply:
@devin-petersohn now the script output is like this, see above.

Reply:
Looks great!